app/src/main/java/io/github/danthe1st/jdoc4droid/activities/list/classes/ListClassesActivity.java

```diff
@@ -72,6 +76,31 @@ public void onCreate(Bundle savedInstanceState) {
         });
     }
 
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        boolean ret=super.onCreateOptionsMenu(menu);
+        filterButton=menu.findItem(R.id.app_bar_filter);
+        filterButton.setVisible(true);
+        filterButton.setOnMenuItemClickListener(menuItem -> {
+            filterButton.getSubMenu().clear();
+            for (String availableFilter : getAvailableFilters()) {
```

Comment:
you can move the for-each-loop to another method (e.g. `loadFilters()`) and call it inside `runInUIThread()` inside `getThreadPool().execute()` in `onCreate` like the following:
```java
    @Override
    public void onCreate(Bundle savedInstanceState) {
        setContentView(R.layout.activity_list_classes_list);
        super.onCreate(savedInstanceState);
        javaDocDir = new File(getIntent().getStringExtra(ARG_JAVADOC_DIR));
        getThreadPool().execute(()-> {
            try {
                descriptions = JavaDocParser.loadClasses(javaDocDir);
                classTypes = descriptions.stream().map(SimpleClassDescription::getClassType).collect(Collectors.toSet());
                runInUIThread(()->{
                    adapter.setItems(descriptions);
                    findViewById(R.id.progressBar2).setVisibility(View.GONE);
                    reloadFilters();//call it here
                });
            } catch (Exception e) {
                runInUIThread(()->{
                    Toast.makeText(this,"Cannot load classes",Toast.LENGTH_LONG).show();
                    onBackPressed();
                    Log.e(getClass().getName(),"Cannot load class list",e);
                });
            }
        });
    }
```
```java
    @UiThread
    private void reloadFilters(){
        for (String availableFilter : getAvailableFilters()) {
            filterButton.getSubMenu().add(availableFilter);
        }
    }
```
This allows to remove the code clearing and re-filling the submenu every time it is clicked.

Reply:
alright makes sense